[Kelvin Soo] iP

KelvinSoo opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER DUPER FAST to use

#### All you need to do is,

1. download it from [here](https://github.com/KelvinSoo/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you

And it is **FREE!**

#### Features

- [x] Managing tasks
- [x] Managing deadlines
- [x] Reminders

#### If you Java programmer, you can use it to practice Java too. Here's the ```main``` method:

```java
public class Duke {
   public static void main(String[] args) {
      new Duke().run();
   }
}
```

Commit:
Add Gradle support

Commit:
Added greet, echo and exit feature to Duke.java

Commit:
Add list funtionality, Change display box design.

Commit:
Add Task.java class.

Commit:
Add Mark as Done functionality.

Commit:
Add ToDos, Events, Deadlines functionality.

Commit:
Add the input/output redirection to semi-automate the testing of the chatbot.

Commit:
Add error handleing.

Commit:
Added more test cases.

Commit:
Add Delete functionality.

Commit:
Add test case for delete.

Commit:
Refactor Duke: change format to better follow coding standard

Commit:
Duke printReply: change the output box design

Commit:
Update EXPECTED.txt

Commit:
Add Save functionality

Commit:
Save function: improve exception handling

Commit:
Task class: convert into abstract class

Commit:
Add Dates and Times functionality

Commit:
Revert commit

Commit:
Revert "Save function: improve exception handling"

This reverts commit e00611ac870a3670dfd346e2e9a6109d25be5626.

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

Commit:
Update version descripting to Level-8

Commit:
Refactor Duke: change design to follow the OOP approach

Commit:
Divide classes into packages

Commit:
Add JUnit tests

Commit:
Package the App as a JAR file

Commit:
Add more JavaDoc comments

Commit:
Tweak code to comply with the coding standard

Commit:
Add Find functionality

Commit:
Revert "Package the App as a JAR file"

This reverts commit 89990239752dfbf5de86b0d59e4fd93188921edc.

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'master' of https://github.com/KelvinSoo/ip

Commit:
Update test files to include testing for find

Commit:
Merge branch 'add-gradle-support' of https://github.com/KelvinSoo/ip

Commit:
Rename methods to be more meaningful

KelvinSoo committed:
Update README.md

JeffZincatz commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+import duke.task.TaskList;
+
+/**
+ * A chatbot based on Project Duke
```
Need punctuation behind each line of header comments, including parameters as well.

JeffZincatz commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    /**
+     * a method to execute a command.
```
Need to capitalize the first letter of the sentence.

JeffZincatz commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,57 @@
+import duke.storage.Storage;
+import duke.ui.Ui;
+import duke.task.Task;
+import duke.task.TaskList;
```
What order of import statements does this use? Need to keep import statement order consistent.

JeffZincatz commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,95 @@
+     * @param taskList The task list to execute the command on
+     * @param ui The user interface to display the reply
+     * @param storage The place to store the session
+     */
```
Need the "@throws" field in Javadocs.

KelvinSoo replied:
I've completely forgotten about it. Thanks for pointing it out for me!

JeffZincatz commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,57 @@
+        String[] command = userCommand.split(" ", 2);
+        DukeException missingParamException =
+                new DukeException(String.format("OOPS!!! Parameter for \"%s\" is missing", command[0]));
+        boolean noParameter = command.length < 2;
```
Naming of boolean variable can be clearer. Maybe `hasNoParameter`?

JeffZincatz commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,96 @@
+    /**
+     * Save task method.
+     * Save the task list at the given file path.
+     * @param taskList The task list to save
```
Need an empty line between description and parameter section.

JeffZincatz reviewed (commented):
In general, not many coding style violations.

For some recurring changes needed, such as empty line between JavaDoc description and parameter, I only commented in one place so that I do not overwhelm you with copies of the same comment.

Commit:
Add Check style configs

Commit:
Merge branch 'master' of https://github.com/KelvinSoo/ip

Commit:
Add gradle gradle dependencies

Commit:
Add basic javaFX feature

Commit:
Add graphical UI

Commit:
Merge branch 'branch-Level-10'

Commit:
Refactor Gui: remove magic numbers

Commit:
Add Assertions for command package

Code should include assertion to alert programmer errors.

Assertions have been added to the AddCommand, DeleteCommand,
DoneCommand, FindCommand and ListCommand.

Using assertion for the command package is important as there might
be instances of programmer error when passing values that are
incompatible such as null list.

Commit:
Refactor DialogBox: remove magic numbers

KelvinSoo committed:
Merge pull request #1 from KelvinSoo/branch-A-Assertions

Branch a assertions

Commit:
Refactor imports

Import statement is not in the right order.

When the import statement is disorganise it will be hard to read.

Organising the import statements by packages increases readability.

Commit:
Refactor JavaDocs

Remove unnecessary white space.

This is to improve code readability.

Commit:
Refactor duke: improve if statements

It is hard to understand some if statements.

Add Boolean conditions with appropriate names.

This is to improve the overall code readability and quality.

KelvinSoo committed:
Merge branch 'master' into branch-A-CodeQuality

Commit:
Add Tagging functionality

Commit:
Merge branch 'branch-BCD-Extension'

# Conflicts:
#	src/main/java/duke/command/DoneCommand.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/TaskList.java

Commit:
Tag command: wrong if statement

Statement is now correct.

Commit:
Refactor UI design from Gui into fxml files

KelvinSoo committed:
Add Ui.png

Commit:
Refactor exceptions into individual duke exception

Commit:
Merge branch 'master' of https://github.com/KelvinSoo/ip

Commit:
Update User Guide

KelvinSoo committed:
Update README.md

Commit:
Update Ui design

Commit:
Improve JavaDocs quality

Commit:
Add main class name to gradle
